zk-token-sdk/src/curve25519/edwards.rs

```diff
@@ -114,6 +117,12 @@ mod target_arch {
         type Point = Self;
 
         fn multiscalar_multiply(scalars: &[PodScalar], points: &[Self]) -> Option<Self> {
+            if scalars.len() > MAX_EDWARDS_MULTISCALAR_MUL
```

Comment:
do these not need gated too?

Reply:
Yeah, I looked to feature gate this logic, but since this is a library function, it wasn't clear how to feature gate this logic. The only way to reach this function though is via the multiscalar mult syscall and this syscall is now feature-gated, so my understanding is that we don't need to feature gate here (please correct me if I am wrong though...)

Reply:
the problem is that this changes the pre-activation behavior of the syscall. this makes testnet and devnet vulnerable to consensus breakage during an upgrade.

since the syscall is already guarded, why do we need to replicate the guards here?

Reply:
Oh yes, you are absolutely right. This changes the behavior before the newly added feature gate is activated. I think it makes sense to remove this condition since this is a library function and the syscall has its own guard.